Core/include/Acts/Vertexing/KalmanVertexTrackUpdater.ipp

```diff
@@ -19,18 +19,13 @@ void Acts::KalmanVertexTrackUpdater::update(TrackAtVertex<input_track_t>& track,
   // Get the linearized track
   const LinearizedTrack& linTrack = track.linearizedState;
 
-  // Check if linearized state exists
-  if (linTrack.covarianceAtPCA.determinant() == 0.) {
-    // Track has no linearized state, returning w/o update
-    return;
-  }
```

Comment:
Should we maybe check `if (track.isLinearized)` here? If the track is not linearized I think that `posJack` etc. are just zero matrices (but I might be wrong).

Reply:
that would definitely make more sense than checking the `det`. I just noticed that we don't check that in the other KF updater and thought it makes sense to align that

Reply:
I think we linearize the track before we use it to update the vertex 
![image](https://github.com/acts-project/acts/assets/72298366/c509f985-c8c6-485c-9c27-5183f124f4ce)